security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -781,66 +792,57 @@ protected HttpResponse executeHttpRequest(CloseableHttpAsyncClient httpclient, H
         return future.get(5, TimeUnit.MINUTES);
     }
 
-    private void copyLocationHeaders(HttpResponse proxiedResponse, HttpServletResponse finalResponse) {
-        for (Header locationHeader : proxiedResponse.getHeaders("Location")) {
-            finalResponse.addHeader(locationHeader.getName(), locationHeader.getValue());
-        }
+    private @Nullable String extractLocationHeader(HttpResponse proxiedResponse) {
+        Header location = proxiedResponse.getFirstHeader("Location");
+        return location == null ? null : location.getValue();
     }
 
-    private void adjustLocation(HttpServletRequest request, HttpResponse proxiedResponse, HttpServletResponse finalResponse) {
+    private Optional<String> adjustLocation(HttpServletRequest request, HttpResponse proxiedResponse) {
         if (logger.isDebugEnabled()) {
```

Comment:
not really. You don't __need__ to do `logger.isDebugEnabled()` at all. It's usually just a safeguard to prevent the performance penalty of building up the log message when the log is performed on a frequently executed code block.
So you probably mean slf4j's ability to use a String pattern and varargs like in `log.debug("unable to connect to {} as user {}", resource, userName)`, which is much better. Yet not to be used wildly especially on very frequently called code, the varargs to array conversion the JVM performs is not for free either. But in general terms, yes, I prefer using slf4j as the logging API and logback as the logging backend, due to its flexibility and ability to redirect other logging apis (e.g. library dependencies using java.util.logging)